### PR TITLE
Move all tests and test data into tests/ directory

### PR DESCRIPTION
**Pull request by tiran**
Tests are no longer in the same Python file as the implementation. This
makes it easier to split custodia into a client and server package.

Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: tests/test_custodia.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def find_port(host='localhost'):
 [global]
 server_version = "Secret/0.0.7"
 server_url = ${SOCKET_URL}
-auditlog = test_audit.log
+auditlog = tests/tmp/test_audit.log
```
Comment (simo5) [inline tests/test_custodia.py]: if we are going to add directory structure then it should be passed in as ${TESTS_DIR} like we do for SOCKET_URL

File: tests/test_custodia.py [hunk 2]
```diff
@@ -121,18 +121,10 @@ def find_port(host='localhost'):
...
-def unlink_if_exists(filename):
-    try:
-        os.unlink(filename)
-    except OSError as err:
-        if err.errno != errno.ENOENT:
-            raise
-
-
-def generate_all_keys(filename):
+def generate_all_keys(filename, dburi):
```
Comment (simo5) [inline tests/test_custodia.py]: should pass in a testsdir not dburi

File: tests/test_custodia.py (hunk 1 above)
Comment (tiran) [inline tests/test_custodia.py]: SOCKET_URL is required for different sockets (TCP/IP, Unix stream). The test directory is always the same. It's fine to hard code it for tests.

Comment (simo5) [inline tests/test_custodia.py]: I prefer to pass it in, you are changing it now, and I can see the need to change it again in future, a single variable is easier to change, please introduce a testdir variable.

**Comment by tiran** [comment]
custodia tests are now using cls.test_dir and TEST_DIR config var.

**Pull request closed without merging**